distribution/lib/Standard/Visualization/0.0.0-dev/src/Scatter_Plot.enso

```diff
@@ -178,10 +202,20 @@ limit_data limit data = case limit of
         if limit <= extreme.length then extreme.take (..First limit) else
             extreme + data.take (..Sample (limit - extreme.length))
 
+
+## PRIVATE
+add_fields : Builder -> Integer -> Integer -> Vector | Nothing
+add_fields current number_of_numeric idx=0 = if idx == number_of_numeric then current.to_vector else
+    current.append (Point_Data.Y_multi idx)
+    @Tail_Call add_fields current number_of_numeric idx+1
```

Comment:
probably a nitpick, but I'm hesitant to exposing such looping functions as top-level methods. IMO it is usually easier to understand if the definition of such loop-helper is next to it usage - so I'd move it into `json_from_table`.

Reply:
But more importantly, I'm not sure this loop is really needed? I will write a comment below.